commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -32,23 +34,27 @@
     private String coordsLog;
 
     protected static Set<String> categorySet;
+    private static List<String> categoryList;
```

Comment:
I think this comment comes too late but, in the future, I most strongly urge you to avoid static state like this. It's common to hear that "globals are evil". This is effectively a private global. Static architecture limits you to a single global instance which affects threading, unit testing, and reusability. In general, when static is used for state, it greatly limits what's practical to do with the class and quickly requires refactoring. In my opinion, static is good for stateless (immutable) utility classes, singleton instances (with the singleton instance usually preserved in the Application subclass), and constants (which are also stateless).